fastparquet/test/test_writer.py

```diff
@@ -272,6 +283,23 @@ def test_groups_roundtrip(tempdir):
         assert row.b in list(df[(df.a==row.a)&(df.c==row.c)].b)
 
 
+def test_empty_groupby(tempdir):
+    df = pd.DataFrame({'a': np.random.choice(['a', 'b', None], size=1000),
+                       'b': np.random.randint(0, 64000, size=1000),
+                       'c': np.random.choice([True, False], size=1000)})
+    df.loc[499:, 'c'] = True  # no False in second half
+    writer.write(tempdir, df, partition_on=['a', 'c'], file_scheme='hive',
+                 partitions=[0, 500])
+    r = ParquetFile(tempdir)
+    assert r.count == sum(~df.a.isnull())
+    assert len(r.row_groups) == 6
+    out = r.to_pandas()
```

Comment:
They don't exist, but what happens is, you get lots and lots of files; but this is users' choice, if they want that, something very loosely like indexing the data. You could in theory have as many row-groups as you have rows, although that would obviously not be very useful.

As for a test, how many is a good test? 1000 files?